flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -489,9 +487,7 @@ abstract class PlannerBase(
       TimeZone.getTimeZone(tableConfig.getLocalTimeZone).getOffset(epochTime)
     tableConfig.set(TABLE_QUERY_START_LOCAL_TIME, localTime)
 
-    getExecEnv.configure(
-      configuration,
-      Thread.currentThread().getContextClassLoader)
+    getExecEnv.configure(tableConfig.getConfiguration, Thread.currentThread().getContextClassLoader)
```

Comment:
add a comment that we intentionally don't want to repropagate the rootConfigutation here and in the `DefaultExecutor`